eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/BundleBuilder.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.core.Box;
+import com.schedjoules.eventdiscovery.framework.serialization.core.Boxable;
+import com.schedjoules.eventdiscovery.framework.serialization.core.FluentBuilder;
+import com.schedjoules.eventdiscovery.framework.serialization.core.Key;
+
+import org.dmfs.optional.Optional;
+
+
+/**
+ * {@link FluentBuilder} for {@link Bundle}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class BundleBuilder implements FluentBuilder<Bundle>
+{
+    private final Bundle mBundle;
+
+
+    public BundleBuilder(Bundle bundle)
+    {
+        mBundle = new Bundle(bundle);
+    }
+
+
+    public BundleBuilder()
+    {
+        mBundle = new Bundle();
+    }
+
+
+    private BundleBuilder(Bundle bundle, String nonUsedParam)
```

Comment:
I'd prefer a slightly different pattern to have this internal non-cloning constructor.
Consider this:

```
   public BundleBuilder() 
   {
       // we don't need to clone Bundle.EMPTY, because we don't try to mutate it
       this(Bundle.EMPTY, false);
   }

   public BundleBuilder(Bundle bundle) 
   {
       this(bundle, true);
   }

   private BundleBuilder(Bundle bundle, boolean clone)
   {
      mBundle = clone ? new Bundle(bundle) : bundle;
   } 
```